src/main/java/address/controller/RootLayoutController.java

```diff
@@ -71,9 +127,18 @@ private void handleNew() {
     @FXML
     private void handleOpen() {
         // Show open file dialog
-        File file = getXmlFileChooser().showOpenDialog(mainController.getPrimaryStage());
-        if (file == null) return;
-        EventManager.getInstance().post(new LoadDataRequestEvent(file));
+        File toOpen = getXmlFileChooser().showOpenDialog(mainController.getPrimaryStage());
+        if (toOpen == null) return;
+        PrefsManager.getInstance().setSaveLocation(toOpen);
+        EventManager.getInstance().post(new LoadDataRequestEvent(toOpen));
+    }
+
+    @FXML
+    private void handleChooseMirror() {
+        // Show open file dialog
+        File toSyncWith = getXmlFileChooser().showOpenDialog(mainController.getPrimaryStage());
+        if (toSyncWith == null) return;
```

Comment:
Hmm but doing that means we cannot access the file chooser and do things like set the starting directory. I think we can just catch `null` here, since it's caused by external libraries. Or is there a better way?